src/content/app/entity-viewer/state/bookmarks/entityViewerBookmarksSlice.ts

```diff
@@ -40,6 +47,77 @@ type UpdatePreviouslyViewedPayload = {
   };
 };
 
+export const updatePreviouslyViewedEntities = createAsyncThunk(
+  'entity-viewer/updatePreviouslyViewedEntities',
+  (params: UpdatePreviouslyViewedPayload, thunkAPI) => {
+    const { genomeId, gene } = params;
+    const { getState } = thunkAPI;
+    const state = getState() as RootState;
+
+    const previouslyViewedEntities = getPreviouslyViewedEntities(
+      state,
+      genomeId
+    );
+
+    const alreadySavedGene = previouslyViewedEntities.find(
+      (entity) => entity.entity_id === gene.unversioned_stable_id
+    );
+
+    const newPreviouslyViewedEntity = {
+      entity_id: gene.unversioned_stable_id,
+      label: gene.symbol ? [gene.symbol, gene.stable_id] : gene.stable_id,
+      type: 'gene' as const
+    };
+
+    const updatedEntities = alreadySavedGene
+      ? previouslyViewedEntities
+      : [newPreviouslyViewedEntity, ...previouslyViewedEntities];
+
+    // side effect
+    entityViewerBookmarksStorageService.updatePreviouslyViewedEntities({
+      [genomeId]: updatedEntities
+    });
+
+    return {
+      genomeId,
+      updatedEntities
+    };
+  }
+);
+
+export const updatePreviouslyViewedEntity = createAsyncThunk(
+  'entity-viewer/updatePreviouslyViewedEntity',
+  (
+    params: {
+      genomeId: string;
+      previouslyViewedEntityId: string;
+      fragment: Partial<PreviouslyViewedEntity>;
+    },
+    thunkAPI
+  ) => {
+    const { genomeId, previouslyViewedEntityId, fragment } = params;
+    const state = thunkAPI.getState() as RootState;
+    const updatedEntities = getPreviouslyViewedEntities(state, genomeId).map(
+      (entity) => {
+        if (entity.entity_id === previouslyViewedEntityId) {
+          return merge({}, entity, fragment);
+        }
+        return entity;
+      }
+    );
+
+    // side effect
+    entityViewerBookmarksStorageService.updatePreviouslyViewedEntities({
+      [genomeId]: updatedEntities
+    });
+
+    return {
+      genomeId,
+      updatedEntities
+    };
+  }
+);
```

Comment:
I remember we added this thunk in Ridwan's PR in order to be able to talk to it from `geneViewTranscriptsSlice`. It doesn't seem to be used anywhere now, does it?